tests/server/test_logging.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import absolute_import, unicode_literals
+
+import mock
+import threading
+import unittest
+
+from pysoa.server.logging import PySOALogContextFilter
+
+
+class TestPySOALogContextFilter(unittest.TestCase):
+    def test_threading(self):
+        thread_data = {}
+
+        def fn(*_, **__):
+            thread_data['first_get'] = PySOALogContextFilter.get_logging_request_context()
+
+            PySOALogContextFilter.set_logging_request_context(foo='bar', **{'baz': 'qux'})
+
+            thread_data['second_get'] = PySOALogContextFilter.get_logging_request_context()
+
+            if thread_data.get('do_clear'):
+                PySOALogContextFilter.clear_logging_request_context()
+
+            thread_data['third_get'] = PySOALogContextFilter.get_logging_request_context()
+
+        self.assertIsNone(PySOALogContextFilter.get_logging_request_context())
+
+        PySOALogContextFilter.set_logging_request_context(request_id=1234, **{'correlation_id': 'abc'})
+
+        self.assertEqual(
+            {'request_id': 1234, 'correlation_id': 'abc'},
+            PySOALogContextFilter.get_logging_request_context()
+        )
+
+        thread = threading.Thread(target=fn)
+        thread.start()
+        thread.join()
+
+        self.assertEqual(
+            {'request_id': 1234, 'correlation_id': 'abc'},
+            PySOALogContextFilter.get_logging_request_context()
+        )
+
+        self.assertIsNone(thread_data['first_get'])
+        self.assertEqual({'foo': 'bar', 'baz': 'qux'}, thread_data['second_get'])
+        self.assertEqual({'foo': 'bar', 'baz': 'qux'}, thread_data['third_get'])
+
+        thread_data['do_clear'] = True
+
+        thread = threading.Thread(target=fn)
```

Comment:
I don't like idea of spawning real threads in tests, but I guess as long as you don't put any asserts in there, it ought to be fine...

Reply:
No other real way to test that it works properly under threads. :-)